macros/src/bundle.rs

```diff
@@ -50,7 +50,7 @@ fn gen_dynamic_bundle_impl(
             }
 
             fn type_info(&self) -> ::std::vec::Vec<::hecs::TypeInfo> {
```

Comment:
Should this also become `with_type_info` for symmetry and to permit memory reuse in e.g. `EntityBuilder`?

Reply:
I have added another commit doing that change, but I do not immediately see how the memory reuse in `EntityBuilder` would look like?

Reply:
Do you just want to memoize the collected vectors? Otherwise, I think this is a net loss w.r.t. allocation as it is.

Reply:
My thinking was that `entity_builder::Common` and `CommandBuffer` would own a `Vec<TypeInfo>` that they'd clear and reuse, avoiding reallocs when those structures are reused as recommended. The only time we call this and don't expect a `Vec` out is in `get_insert_target`, so this probably doesn't matter much, but if we can avoid allocating a fresh `Vec` in the implementations of `type_info` then that should save an alloc per insert on average.

Reply:
> but if we can avoid allocating a fresh Vec in the implementations of type_info then that should save an alloc per insert on average.

But only for the slow path for insertion as otherwise the `InsertTarget` is already cached, right? Whereas in `spawn_inner` we would still need to clone even the re-used `Vec`.

After seeing it implemented, I think I would personally prefer to drop that last commit.

Reply:
Right, that makes sense; let's drop it.

Reply:
Done.